ext/storage_sources/gcp_store/test/test_gcp_connection.cpp

```diff
@@ -30,5 +30,338 @@
 #include <catch2/catch.hpp>
 
 #include "gcp_connection.h"
+#include <typeinfo>
 
-TEST_CASE("Testing class gcpConnection", "gcp-connection") {}
+namespace gcs = google::cloud::storage;
+using namespace gcs;
+namespace test_defaults {
+static std::string bucket_name("unit_testing_gcp");
+
+/*
+ * Objects with the prefex pattern "gcptest/*" are deleted after a certain period of time according
+ * to the lifecycle rule on the gcp bucket. Should you wish to make any changes to the prefix
+ * pattern or lifecycle of the object, please speak to the release manager.
+ */
+static std::string obj_prefix("gcptest"); // To be concatenated with a random string.
+} // namespace test_defaults
+
+// Concatenates a random suffix to the prefix being used for the test object keys. Example of
+// generated test prefix: "gcptest/unit/2022-31-01-16-34-10/623843294--".
+static int
+randomize_test_prefix()
+{
+    char time_str[100];
+    std::time_t t = std::time(nullptr);
+    std::string new_prefix = "gcptest-unit-";
+
+    REQUIRE(std::strftime(time_str, sizeof(time_str), "%F-%H-%M-%S", std::localtime(&t)) != 0);
+
+    new_prefix += time_str;
+
+    // Create a random device and use it to generate a random seed to initialize the generator.
+    std::random_device my_random_device;
+    unsigned seed = my_random_device();
+    std::default_random_engine my_random_engine(seed);
+
+    new_prefix += '/' + std::to_string(my_random_engine());
+    new_prefix += "--";
+
+    test_defaults::obj_prefix = new_prefix;
+
+    return (0);
+}
+
+// Overrides the defaults with the ones specific for this test instance.
+static int
+setup_test_defaults()
+{
+    // Append the prefix to be used for object names by a unique string.
+    REQUIRE(randomize_test_prefix() == 0);
+    // std::cerr << "Generated prefix: " << test_defaults::obj_prefix << std::endl;
+
+    return (0);
+}
+
+TEST_CASE("Testing class gcpConnection", "gcp-connection")
+{
+    // Setup the test environment.
+    REQUIRE(setup_test_defaults() == 0);
+
+    gcp_connection conn(test_defaults::bucket_name, test_defaults::obj_prefix);
+    // bool exists = false;
+    // size_t objectSize;
+
+    const std::string object_key = "test_object";
+    const std::string file_name = "test_object.txt";
+    const std::string file_path = "./" + file_name;
+    const std::string non_exi_object_key = "test_non_exist";
+    const std::string non_exi_file_name = "test_non_exist.txt";
+    const bool list_single = true;
+    std::vector<std::string> objects;
+
+    google::cloud::storage::Client client = conn.get_client();
+
+    std::ofstream File(file_name);
+    std::string payload = "Test payload :)";
+    File << payload;
+    File.close();
+
+    SECTION("Simple list test", "[gcp-connection]")
+    {
+        const int32_t total_objects = 20;
+
+        // No matching objects. Object size should be 0.
```

Comment:
This comment is a bit misleading. I believe what you mean is that the `objects` array is empty.